uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Document.php

```diff
@@ -38,118 +38,115 @@ public static function create(&$params) {
+      'is_deleted' => 0,
+      'status_id' => self::STATUS_APPROVED,
+      'options' => ['limit' => 0],
+      'return' => ["target_contact_id", "activity_type_id", "details"]
+    ];
+
+    $result = civicrm_api3('Document', 'get', $params);
+
+    return CRM_Utils_Array::value('values', $result, []);
   }
 
   /**
-   * Search for approved Documents having expire date less than given $expireDate
-   * and create their clone.
-   * Return a count of cloned documents.
-   * 
-   * @param DateTime $expireDate
-   * @return int
+   * Creates a copy of a document with no attached files
+   *
+   * @param array $original
    */
-  protected static function cloneApprovedDocumentsWithExpireDate(DateTime $expireDate) {
-    $count = 0;
-    $query = "SELECT ac.id, ac.entity_id FROM civicrm_value_activity_custom_fields_11 ac "
-      . "LEFT JOIN civicrm_component c ON c.name = 'CiviDocument' "
-      . "LEFT JOIN civicrm_option_group og ON og.name = 'activity_type' "
-      . "LEFT JOIN civicrm_option_value ov ON ov.option_group_id = og.id AND ov.component_id = c.id "
-      . "LEFT JOIN civicrm_activity a ON a.id = ac.entity_id "
-      . "WHERE DATE(ac.expire_date) <= %1 AND ac.clone_date IS NULL AND a.is_deleted = 0 AND a.status_id = 3 AND ov.component_id = c.id AND a.activity_type_id = ov.value";
-    $params = array(
-      1 => array($expireDate->format('Y-m-d'), 'String'),
-    );
-    $result = CRM_Core_DAO::executeQuery($query, $params);
-    while ($result->fetch()) {
-      $document = civicrm_api3('Document', 'getsingle', array(
-        'id' => $result->entity_id,
-        'return' => 'activity_type_id, activity_date_time, details, status_id, is_deleted, expire_date, case_id, source_contact_id, target_contact_id, assignee_contact_id',
-      ));
-      $clone = self::cloneDocument($document, $result->id);
-      if (!empty($clone['id'])) {
-        $count++;
-      }
+  protected static function cloneDocument(array $original) {
+    $clone = $original;
+    $origId = $original['id'];
+
+    $clone['activity_date_time'] = $original['expire_date'];
+    $clone['status_id'] = self::STATUS_AWAITING_UPLOAD;
+
+    $fieldsToUnset = ['id', 'file_count', 'expire_date', 'document_number', 'valid_from'];
+    foreach ($fieldsToUnset as $field) {
+      unset($clone[$field]);
     }
-    return $count;
+
+    civicrm_api3('Document', 'create', $clone);
+
+    // Update original's clone_date' so we know it has been cloned.
+    $today = (new DateTime())->format('Y-m-d');
+    $params = ['id' => $origId, 'clone_date' => $today];
+    civicrm_api3('Document', 'create', $params);
   }
 
   /**
-   * Clone particular document but with unsetting some fields according to
-   * PCHR-1365 requirements.
-   * 
-   * @param array $document
-   * @param int $expireId
-   * @return array
+   * Gets the custom field name for Documents
+   *
+   * @param string $fieldName
+   *  The human-readable name for the field, e.g. "expiry_date"
+   * @return mixed|null
```

Comment:
It's just expecting the same return value as `CRM_Utils_Array::value`, which is `mixed`, it was autogenerated and since it doesn't complain if I use `string` I'll change it.